### PR TITLE
Publish date extension

### PR DESCRIPTION
**Pull request by thecalcc**
BOR-21

### CODE REVIEW

File: scripts/apps/authoring/authoring/modified-info.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export class ModifiedInfo extends React.PureComponent<IProps> {
     render() {
         const {article} = this.props;
 
-        if (article.versioncreated == null) {
+        if (article.versioncreated == null || article.pubstatus === 'CANCELED') {
```
Comment (MarkLark86) [inline scripts/apps/authoring/authoring/modified-info.tsx]: What is the reason to now show the versioncreated if the item has been killed?

File: scripts/extensions/topbar-publish-date/src/publish-date.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React = require('react');
+import {IArticle, ISuperdesk} from 'superdesk-api';
+
+export function getPublishDate(superdesk: ISuperdesk) {
```
Comment (MarkLark86) [inline scripts/extensions/topbar-publish-date/src/publish-date.tsx]: When an extension is registered, the `superdesk` API is populated on the window object. Which means you don't need this wrapper function, you can simply add this at the top of the file `import {superdesk} from './superdesk'`

File: scripts/extensions/topbar-publish-date/src/publish-date.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+            if (article.firstpublished == null || article.pubstatus.toLowerCase() === 'canceled') {
+                return null;
+            } else {
+                return (
+                    <dl data-test-id="date-published">
+                        <dt>{superdesk.localization.gettext('Published')}</dt>
+                        {' '}
+                        <dd>
+                            <DateTime dateTime={article.firstpublished} />
```
Comment (MarkLark86) [inline scripts/extensions/topbar-publish-date/src/publish-date.tsx]: Is the date/time for the first version published, or the latest version published?

File: scripts/extensions/topbar-publish-date/src/publish-date.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React = require('react');
+import {IArticle, ISuperdesk} from 'superdesk-api';
+
+export function getPublishDate(superdesk: ISuperdesk) {
+    return class DisplayPublishedTime extends React.PureComponent<{article: IArticle}> {
+        render() {
+            const {article} = this.props;
+            const {DateTime} = superdesk.components;
+
+            if (article.firstpublished == null || article.pubstatus.toLowerCase() === 'canceled') {
```
Comment (MarkLark86) [inline scripts/extensions/topbar-publish-date/src/publish-date.tsx]: Again here, if the item is killed why not show the date/time is was killed?

File: scripts/apps/authoring/authoring/modified-info.tsx (hunk 1 above)
Comment (thecalcc) [reply scripts/apps/authoring/authoring/modified-info.tsx]: mistake, forgot to commit the fix

File: scripts/extensions/topbar-publish-date/src/publish-date.tsx (hunk 4 above)
Comment (thecalcc) [reply scripts/extensions/topbar-publish-date/src/publish-date.tsx]: Can be done, just not a requirement from Ivan

File: scripts/extensions/topbar-publish-date/src/publish-date.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React = require('react');
+import {IArticle} from 'superdesk-api';
+import {superdesk} from './superdesk';
+
+export function getPublishDate() {
```
Comment (MarkLark86) [inline scripts/extensions/topbar-publish-date/src/publish-date.tsx]: No need to use a function now. You can just export the class component

**Review by MarkLark86 — APPROVED** [review]

**Review by tomaskikutis — COMMENTED** [review]
Move this to a customer repo.

**Comment by thecalcc** [comment]
> Move this to a customer repo.

I have, check it out here - https://github.com/superdesk/superdesk-borsen/pull/63

**Comment by tomaskikutis** [comment]
I see, let's close this one then

**Pull request closed without merging**